### ISSUE TITLE
ld Errors When Building Agent For Arch Linux

### ISSUE THREAD

**Issue by GraysonPeddie** [wazuh/wazuh#5162]
I wanted to install Wazuh Agent in Arch Linux. After executing an install.sh script, the make gave me an error:

```
 $ cd wazuh && make -C src deps
make: Entering directory '/home/grayson/Downloads/git/wazuh/src'
curl -so external/cJSON.tar.gz https://packages.wazuh.com/deps/3 12./cJSON.tar.gz
make: *** [Makefile:904: external/cJSON.tar.gz] Error 6
make: Leaving directory '/home/grayson/Downloads/git/wazuh/src'
```

I had to make some changes to the Makefile for deps, so I commented out the "VERSION" that had a bug and created a new line as follows:

```
################################
#### External dependencies  ####
################################

TAR := tar -xf
GUNZIP := gunzip
CURL := curl -so
space =
space +=
VERSION = 3.12
#VERSION = $(subst $(space),.,$(wordlist 1,2,$(subst ., ,$(patsubst v%,%,$(shell cat VERSION)))))
```

I also had to manually download cJSON.tar.gz because it when I do a `make -C src deps`, the cJSON.tar.gz file did not get overwritten and I then decompress the file in the external directory.

I then follow instructions made by @BraulioV, dated November 13, 2019, by applying patches for OpenSSL, and did a `make -C src TARGET=agent`.

https://github.com/wazuh/wazuh/issues/3012

However, I ran into an issue here:

```
    LINK libwazuh.a
ar: `u' modifier ignored since `D' is the default (see `U')
    RANLIB libwazuh.a
    CC libwazuhext.so
/usr/bin/ld: missing --end-group; added as last command line option
    CC ossec-agentd
/usr/bin/ld: client-agent/config.o:(.bss+0x4): multiple definition of `rotate_log'; client-agent/agentd.o:(.bss+0x0): first defined here
/usr/bin/ld: client-agent/main.o:(.bss+0x0): multiple definition of `agent_debug_level'; client-agent/config.o:(.bss+0x0): first defined here
/usr/bin/ld: libwazuh.a(msgs.o):(.bss+0x0): multiple definition of `agt'; client-agent/config.o:(.bss+0x10): first defined here
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:1227: ossec-agentd] Error 1
make[1]: Leaving directory '/home/grayson/Downloads/git/wazuh/src'
make: *** [Makefile:603: agent] Error 2
make: Leaving directory '/home/grayson/Downloads/git/wazuh/src'
```

3.12.3 is the version of wazuh that I cloned from the GitHub repository.

Let me know if I can provide any assistance if the bug report is incomplete. Or is this not a bug but an issue in my Arch Linux system?

**Comment by BraulioV** [comment]
Hi @GraysonPeddie,

Thanks for your feedback and your help :) (ArchLinux community is awesome) but we are working to solve these issues that you are experiencing. First, the OpenSSL patch is no longer needed since it was solved some months ago by changing the OpenSSL version to a newer one. 

The other issues that you described on your comment have been located and we are working to solve them. Here you can more details about these issues and the PR that fix them.
- [Makefile is not detecting proper VERSION and failing to download deps](https://github.com/wazuh/wazuh/issues/4773): this issue was reported by a user and it is fixed here https://github.com/wazuh/wazuh/pull/4774. Later, a teammate reported on https://github.com/wazuh/wazuh/issues/5021 that this bug is related to a backward compatibility issue.
- [Global variable compilation bug in GCC 10.1.0](https://github.com/wazuh/wazuh/issues/5104) we opened this issue to report the multiple definition errors that you can see when `ld` is trying to link each component. This is solved here: https://github.com/wazuh/wazuh/pull/5119 and it is being reviewed.
- [Fix CPython's compilation issues on ArchLinux](https://github.com/wazuh/wazuh/pull/5120):  As you can see here https://github.com/wazuh/wazuh/issues/3012, to build the manager successfully it was necessary to build the manager using: `LIBS="-lnghttp2 -lbrotlidec -lrtmp -lpsl" make -C src TARGET=server`. This PR fixes that, making it easier to build the manager.

We are planning to introduce these fixes as soon as possible once they're fully reviewed and approved.

Anyway, if you want to contribute to this project you can create a fork and open a PR. To optimize the review and merge process it is better to first open an issue, then fork the repository and finally open a PR.

If you need any help with the code or with the issue itself, you can ping us on there or ask for help in our [slack channel](https://wazuh.com/community/join-us-on-slack/). All the Wazuhdevelopers are there so one of us can help you.

Sorry for the long answer but I hope it helps.

Regards.

**Comment by GraysonPeddie** [comment]
I suppose I could have checked the list of issues before I report mine. I found #3012 when I was having trouble building the agent and decided to create a new issue.

Anyway, thanks for letting me know that the issues are being fixed. I look forward to installing the agent once the issues are solved.

**Comment by GraysonPeddie** [comment]
I do have a question unrelated to the particular issue: What is the meaning of "community" next to the "bug?" I hovered over the "community" highlight and the tooltip says "community" while "bug" says "something isn't working."

**Comment by BraulioV** [comment]
Hi @GraysonPeddie, 

sorry for the late response. Those are labels that we used to organized the issues. For example, using the label `community` we can quickly get all the issues opened by the community.

This helps us to schedule community requests. This also applies for `bug` label. 

![image](https://user-images.githubusercontent.com/9081114/84382501-05220000-abeb-11ea-8e6d-1036d7477983.png)

Also, I just wanted to mention that we merged https://github.com/wazuh/wazuh/pull/5120 into 3.13 branch. On the other hand, https://github.com/wazuh/wazuh/issues/4773 is still under-review but, if you want to give it a try, you can build the agent or the manager as follows:

```shellsession
$ git clone https://github.com/wazuh/wazuh -b 3.13 && cd wazuh
$ git merge origin/5104-Fix-compilation-bug-GCC-10 --no-commit
$ make -C src deps RESOURCES_URL=https://packages.wazuh.com/deps/3.13
$ make -C src TARGET=server -j $(nproc) # for the manager
$ make -C src TARGET=agent -j $(nproc) # for the agent
# ./install.sh
```

**Comment by GraysonPeddie** [comment]
Okay, thanks.

**Comment by BraulioV** [comment]
Hi again @GraysonPeddie,

I'm closing this issue because all the PRs have been merged. To give you more details, all the fixes are available in the `3.14` branch. We will release v3.13.0 very soon and the code is frozen and the changes applied to fix https://github.com/wazuh/wazuh/issues/5104 are quite important, so they have been scheduled to the next version.

For now, you can install it on ArchLinux as follows:

```shellsession
$ git clone https://github.com/wazuh/wazuh -b 3.14 && cd wazuh
$ make -C src deps 
$ make -C src TARGET=server -j $(nproc) # for the manager
$ make -C src TARGET=agent -j $(nproc) # for the agent
# ./install.sh
```

Or build .deb and .rpm packages using the tools from [wazuh-packages](https://github.com/wazuh/wazuh-packages). More info at: https://documentation.wazuh.com/3.12/development/packaging/index.html

Thanks again for your feedback and feel free to re-open the issue, give us feedback or collaborate with us.
